### ISSUE TITLE
How to quantized AlexNet?

### ISSUE THREAD

**Issue by RichardMrLu** [TropComplique/trained-ternary-quantization#4]
I changed the net to AlexNet(original alexnet, no batchnorm,but have dropout), and changed dataset. I follow the paper, the first conv and the last fc3 ,their params are not quantized.

HYPERPARAMETER_T = 0.15
LEARNING_RATE = 1e-1

all_kernels are tensor that only contain 0、1、-1
initial_scaling_factors are (1,1)....

but training:
0  5348670964449068.000 4.063  0.048 0.059  0.218 0.264  148.051
1  11028.571 3.509  0.055 0.057  0.271 0.278  143.362
2  11.255 3.473  0.053 0.055  0.273 0.283  141.120
3  35.804 3.465  0.055 0.057  0.271 0.280  144.584
4  41.598 3.470  0.056 0.055  0.272 0.271  133.428
5  10366.085 3.469  0.055 0.059  0.268 0.280  133.706
Epoch     5: reducing learning rate of group 0 to 1.0000e-02.
Epoch     5: reducing learning rate of group 1 to 1.0000e-02.
Epoch     5: reducing learning rate of group 0 to 1.0000e-02.
Epoch     5: reducing learning rate of group 0 to 1.0000e-02.
6  23.033 3.464  0.057 0.059  0.274 0.280  133.508
....

the accuracy is so low, it seems params didn't update.

**Comment by RichardMrLu** [comment]
..this probem is solved, thanks for your work.

**Comment by TropComplique** [comment]
Hi! I had the same problem.
You just need a lot of hyperparameter tuning and luck :)
